staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -489,6 +509,10 @@ func (qs *queueSet) removeTimedOutRequestsFromQueueLocked(queue *queue, fsName s
 			metrics.AddRequestsInQueues(req.ctx, qs.qCfg.Name, req.fsName, -1)
 			req.NoteQueued(false)
 		} else {
+			// question: because the queue is ordered with request arrival time
+			// (not in order of their deadline), request(s) further down
+			// in the queue may have already gotten a 504 from the upstream
+			// timeout filter for a shorter deadline than requestWaitLimit
```

Comment:
I think `removeTimedOutRequestsFromQueueLocked` can be deleted and no longer be called by `timeoutOldRequestsAndRejectOrEnqueueLocked` --- whose functionality and name are thereby reduced.

The sad thing is that the time complexity of `cancelWait` is O(number of requests waiting in queue).  We could change that to O(1) by switching from a slice to a linked list.  I suspect we can leave this unchanged until the revision to achieve max-min fairness.